mindmeld/components/entity_resolver.py

```diff
@@ -76,10 +76,11 @@ def __init__(
         self._es_host = es_host
         self._es_config = {"client": es_client, "pid": os.getpid()}
 
-        num_canonical_entities = len(
-            self._resource_loader.get_entity_map(self.type).get("entities", [])
-        )
-        self._no_canonical_entities = num_canonical_entities == 0
+        if self._is_system_entity:
+            canonical_entities = []
+        else:
+            canonical_entities = self._resource_loader.get_entity_map(self.type).get("entities", [])
+        self._no_canonical_entities = len(canonical_entities) == 0
```

Comment:
does this property apply to system entities? @serapio  @arushir

Reply:
System entities dont have canonical maps since they are resolved externally (through Duckling), hence `_no_canonical_entities ` would be `0` for them. The logic seems correct.

Reply:
should this property be None instead of 0 since canonical entities are not applicable to them?

Reply:
The variable is named `_no_canonical_entities`, which is basically a True or False question, so I think its valid for it to be False for system entities since they do not have canonical entities.

Reply:
Yes, there is no entity map for system entities since the resolution is done via Duckling. Maybe we could update this variable to `no_canonical_entity_map` to make this more clear. The logic is good.